### DISCUSSION TITLE
What about splitting the main script into multiple files?

### DISCUSSION THREAD

**Discussion by resucutie** [ThioJoe/YT-Spammer-Purge#288] [Q&A] [answered]
The main script (the one that executes the code) is a very gigantic strip of code. So well, can't it be split into multiple files? Yeah it will take some time, but IMO (i respect if yours is different) it will be worth considering that for most of people can work better with separated files rather than a big one

**Comment by ThioJoe** [comment, marked as answer]
Yes, it has become an absolute behemoth. I am definitely planning on breaking it up relatively soon, as it's becoming more and more difficult even for me to navigate.  I never really expected it to reach this size.

**Reply by resucutie** [reply]
Oh good. As a programmer I know it will be very good for both the project and for you, trust me. I am a plugin developer for BetterDiscord, and well I use a bundler called BDBuilder to bundle all of my files into one single file, because well, there is a issue with BetterDiscord's plugins, becuase they're literally a single JavaScript file to be loaded, and BetterDiscord doesn't allow or have any functionality for mutiple filing, so you see how painful is to make. I am not saying to use a bundler by any means, the project can work without a bundler

**Reply by Lumioze** [reply]
ye this is a good idea, tried looking at the code and used a profiler on it (pyinstrument since i hate cProfile) and got confused because this code is huge as hell like what the hell where do the regexes come from, so far i've only replaced the built in python re with pyre2 (basically a much faster implementation of re in C++)

**Comment by ThioJoe** [comment]
In the latest 2.11 Beta branch I split up the files

**Reply by resucutie** [reply]
Oooo~ thats epic!
